explainer-beforematch.md

```diff
@@ -0,0 +1,135 @@
+# The `beforematch` event
+
+### Summary (TL;DR)
+
+The `beforematch` event is fired at render timing in the following situations:
+
+* There is a new find-in-page active match, in which case the event is fired
+  with the matched element as the target.
+* There is a fragment-link navigation, in which case the event is fired with
+  the matched fragment as the target.
+* There is a scroll-to-text navigation, in which case the event is fired with
+  the matched element as the target.
+
+### Motivation
+
+With the evolution of the web, there are always new and interesting ways that
+developers choose to organize the information on their pages. Some of these
+approaches (e.g. the common case of text scrolling), yield themselves naturally to
+user-agent features like find-in-page. This is not an accident, since
+find-in-page was designed with common use-cases in mind.
+
+However, there are other interesting approaches, such as collapsed sections of
+text, do not work well with user-agent features since the page does not get any
+indication that the user initiated a find-in-page request, or fragment or
+scroll-to-text navigation.
+
+The `beforematch` event is a signal that is a step in the direction that allows
+us to leverage information that the user-agent has to assist the page in making
+the user experience great. Specifically, with future additions to
```

Comment:
Suggest deleting the second sentence

Reply:
Because it de-emphasizes the independent nature of this feature. It has value for collapsed sections already, irrespective of display locking. In fact, the parters who are testing out this use case in our OT don't anticipate any benefit to their use case of the performance aspect of display locking, just the part about discoverability in UA algorithms.

Reply:
Just chime in here. In the Motivation section of the readme, it is mentioned that

> For these reasons, web developers need ways to reduce loading and rendering time of web apps that have a lot of non-visible DOM.

However @chrishtr mentioned above that

> In fact, the parters who are testing out this use case in our OT don't anticipate any benefit to their use case of the performance aspect of display locking

It seems a bit contradictory to me. Are we implying that renderSubtree is focusing more on solving the discoverability problem? 

P.S I am an OT user and yes I didn't observe much performance improvement after applying the display locking to our long list

Reply:
I think @chrishtr comment is about the beforematch event specifically. Display locking addresses a couple of use-cases, one being performance and another being discoverability. When it comes to the beforematch event, the important part of display locking is discoverability.

I'm disappointed to hear that you didn't observe performance benefits from display locking. If you don't mind and have the time to do it, can you file an issue in this repo with some details of how you used it, what you expected to see, and what happened instead? This is an important use-case for us, and we'd like to understand how we can improve performance.

It's also possible that there may be something we overlooked in the implementation which prevents you from seeing the performance results. In either case, the more information you can give us, the better